api/adaptive_load/adaptive_load.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/benchmark_result.proto";
+import "api/adaptive_load/metrics_plugin.proto";
+import "api/adaptive_load/metric_spec.proto";
+import "api/adaptive_load/step_controller.proto";
+import "api/client/options.proto";
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+import "validate/validate.proto";
+
+// Parameters describing the adjusting and testing stages of an adaptive load
+// session, which consists of a series of Nighthawk benchmarks probing for
+// the optimal load on the system, followed by a longer benchmark to validate
+// the values. Load adjustments are calculated by the selected StepController
+// plugin. Metrics can come from Nighthawk stats and counters or
+// platform-specific data sources via MetricsPlugins.
+message AdaptiveLoadSessionSpec {
```

Comment:
For all the fields meant to be provided by the user- can we add an explicit comment whether a field is considered Optional or Required?

Reply:
Done.